orc8r/cloud/go/services/bootstrapper/servicers/registration/registration.go

```diff
@@ -0,0 +1,101 @@
+package registration
+
+import (
+	"context"
+	"fmt"
+
+	"magma/orc8r/cloud/go/orc8r"
+	"magma/orc8r/cloud/go/serdes"
+	"magma/orc8r/cloud/go/services/bootstrapper"
+	"magma/orc8r/cloud/go/services/device"
+	models2 "magma/orc8r/cloud/go/services/orchestrator/obsidian/models"
+	"magma/orc8r/cloud/go/services/tenants"
+	"magma/orc8r/lib/go/protos"
+
+	"github.com/go-openapi/strfmt"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type registrationServicer struct {
+}
+
+func NewRegistrationServicer() (protos.RegistrationServer, error) {
+	return &registrationServicer{}, nil
+}
+
+func (rs *registrationServicer) Register(c context.Context, request *protos.RegisterRequest) (*protos.RegisterResponse, error) {
+	nonce := nonceFromToken(request.Token)
+
+	deviceInfo, err := bootstrapper.GetGatewayDeviceInfo(context.Background(), nonce)
+	if err != nil {
+		return formatRegisterResponseError(
+			fmt.Sprintf("could not get device info from nonce %v: %v", nonce, err),
+		), nil
+	}
+
+	err = registerDevice(*deviceInfo, *request.Hwid, *request.ChallengeKey)
+	if err != nil {
+		return formatRegisterResponseError(fmt.Sprintf("error registering device: %v", err)), nil
+	}
+
+	controlProxy, err := getControlProxy(deviceInfo.NetworkId)
+	if err != nil {
+		return formatRegisterResponseError(fmt.Sprintf("error getting control proxy: %v", err)), nil
+	}
+
+	res := &protos.RegisterResponse{
+		Response: &protos.RegisterResponse_ControlProxy{
+			ControlProxy: controlProxy,
+		},
+	}
+	return res, nil
+}
+
+func registerDevice(ti protos.GatewayDeviceInfo, hwid protos.AccessGatewayID, challengeKey protos.ChallengeKey) error {
```

Comment:
Similar to a comment on another PR -- protos just "want" to be passed around as ptrs, because the grpc-provided iface is always using ptrs. So, let's not fight that. Just pass them around as ptrs and we can remove the addr/de-ref logic